pallets/time-release/src/lib.rs

```diff
@@ -326,16 +376,95 @@ pub mod module {
+		/// * [`Error::MaxReleaseSchedulesExceeded`] - Failed because the maximum release schedules was exceeded-
+		/// * [`ArithmeticError::Overflow] - Failed because of an overflow.
+		///
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::schedule_transfer().saturating_add(T::WeightInfo::transfer()))]
+		pub fn schedule_transfer(
+			origin: OriginFor<T>,
+			dest: <T::Lookup as StaticLookup>::Source,
+			schedule: ReleaseScheduleOf<T>,
+			when: BlockNumberFor<T>,
+		) -> DispatchResult {
+			let from = T::TransferOrigin::ensure_origin(origin)?;
+			let to = T::Lookup::lookup(dest.clone())?;
+
+			let total_amount = Self::validate_and_get_schedule_amount(&schedule)?;
+			Self::ensure_sufficient_free_balance(&from, &to, total_amount)?;
+
+			let _ = Self::update_schedule_vesting_hold(
+				&from,
+				schedule.total_amount().ok_or(ArithmeticError::Overflow)?,
+			);
+
+			Self::schedule_transfer_for(from.clone(), dest, schedule, when)?;
+
+			Ok(())
+		}
+
+		/// Execute a scheduled transfer
+		///
+		/// This function is called to execute a transfer that was previously scheduled.
+		/// It ensures that the origin is valid, the destination account exists, and the
+		/// release schedule is valid. It then finalizes the vesting transfer from the
+		/// hold balance and emits an event indicating that the release schedule was added.
+		///
+		/// # Arguments
+		///
+		/// * `origin` - The origin of the call, which must be the TimeRelease origin.
+		/// * `dest` - The destination account to which the funds will be transferred.
+		/// * `schedule` - The release schedule that specifies the amount and timing of the transfer.
+		///
+		/// # Errors
+		///
+		/// * [`ArithmeticError::Overflow`] - If the total amount in the schedule overflows.
+		/// * [`Error::InsufficientBalanceToFreeze`] - If the hold balance is insufficient.
+		/// * [`Error::ZeroReleasePeriod`] - If the release period is zero.
+		/// * [`Error::ZeroReleasePeriodCount`] - If the release period count is zero.
+		/// * [`Error::AmountLow`] - If the total amount is below the minimum release transfer amount.
+		///
+		/// # Events
+		///
+		/// * [`Event::ReleaseScheduleAdded`] - Indicates that a new release schedule was added.
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::execute_scheduled_transfer())]
+		pub fn execute_scheduled_transfer(
+			origin: OriginFor<T>,
+			dest: <T::Lookup as StaticLookup>::Source,
+			schedule: ReleaseScheduleOf<T>,
+		) -> DispatchResult {
+			let from = T::TimeReleaseOrigin::ensure_origin(origin)?;
+			let to = T::Lookup::lookup(dest)?;
+
+			let total_amount = Self::validate_and_get_schedule_amount(&schedule)?;
+			Self::ensure_sufficient_hold_balance(&from, total_amount)?;
+
+			Self::finalize_vesting_transfer_from_hold(&from, &to, schedule.clone())?;
+
+			Self::deposit_event(Event::ReleaseScheduleAdded {
+				from,
+				to,
+				release_schedule: schedule,
+			});
```

Comment:
I'm confused on why we're emitting this event; didn't we emit this event when the schedule was initially added, and now we're executing a previously-added schedule? Or maybe we just have a naming problem...?

Reply:
Resolution to add documentation.